### PR TITLE
RoboCup Changes: Port ShootGoal Tactic

### LINKED ISSUE #912: RoboCup Changes: Port ShootGoalTactic
### Description of the task

Cleanup and migrate the changes made to the ShootGoalTactic at RoboCup 2019 to the current master branch.

### Acceptance criteria

- [ ] Migrate changes

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description

Changes made at RoboCup 2019

### Testing Done

Updated unit tests

### Resolved Issues

resolves #912 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/ai/hl/stp/tactic/shoot_goal_tactic.cpp [hunk 1]
```diff
@@ -75,19 +75,17 @@ bool ShootGoalTactic::isEnemyAboutToStealBall() const
...
 
-    Point rotated_baller_position = robot->position().rotate(theta);
-
-    Rectangle area_in_front_of_rotated_baller(
-        rotated_baller_position - Vector(0, 3 * ROBOT_MAX_RADIUS_METERS),
-        rotated_baller_position +
-            Vector(5 * ROBOT_MAX_RADIUS_METERS, 3 * ROBOT_MAX_RADIUS_METERS));
+    Rectangle baller_frontal_area = Rectangle(
+        (robot->position() + front_of_robot_dir.perp().norm(ROBOT_MAX_RADIUS_METERS)),
+        robot->position() + front_of_robot_dir.norm(3 * ROBOT_MAX_RADIUS_METERS) -
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/tactic/shoot_goal_tactic.cpp]: Can we explain and/or parameterize the magic number 3?

**Review by jonathanlew — CHANGES_REQUESTED** [review]

File: src/software/ai/hl/stp/tactic/shoot_goal_tactic.cpp (hunk 1 above)
Comment (garethellis0) [reply src/software/ai/hl/stp/tactic/shoot_goal_tactic.cpp]: +1 - I'd probably parameterize `3 * ROBOT_MAX_RADIUS` so we're dealing with an absolute distance rather then a scaling factor, makes it easier to reason about.

**Review by garethellis0 — CHANGES_REQUESTED** [review]

File: src/software/ai/hl/stp/tactic/shoot_goal_tactic.cpp (hunk 1 above)
Comment (MathewMacDougall) [reply src/software/ai/hl/stp/tactic/shoot_goal_tactic.cpp]: changed to a parameter with a description.

**Review by jonathanlew — APPROVED** [review]
LGTM!

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Pull request merged**